src/bin/cli/mod.rs

```diff
@@ -682,3 +697,41 @@ fn explicit_args(matches: &ArgMatches) -> Vec<&Id> {
         })
         .collect()
 }
+
+/// A helper function to check if the target name provided by the user matches the target name in solang.toml
```

Comment:
I like that you wrote this function, thanks. However I think it needs slight refactoring and some tests